### PR TITLE
Eslint plugin: extend gestalt/button-icon-restrictions to cover role=link icon restrictions

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Eslint_plugin: extend gestalt/button-icon-restrictions to cover role=link icon restrictions

#### Why?

Require a specific value when using [Icon](http://localhost:8888/web/icon) with [Button](http://localhost:8888/web/button). Gestalt is more permissive than we recommend internally for adding icons to Buttons, so Buttons using iconEnd must use the "arrow-down" icon and Buttons with link role using iconEnd must use the "visit" icon.

This Eslint rule didn't cover the second restriction, a new best practice that  was recently implemented.

### Codemod

To facilitate the migration, run this codemod to find instances of Button role="link" 

```
yarn codemod detectManualReplacement ~/path/to/your/code
--component=Button
--prop=role
--value=link
```

However, the Eslint rule will through an ERROR/WARNING message anyway that can be used to detect places that need replacement.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4644)

### CODE REVIEW

File: docs/pages/get_started/developers/eslint_plugin.js [hunk 1]
```diff
@@ -88,7 +88,8 @@ With AUTOFIX!
         <MainSection.Subsection
           title="gestalt/button-icon-restrictions"
           description={`
-        Require a specific icon when using an icon with Button. Buttons using \`iconEnd\` must use icon &quot;arrow-down&quot;
+        Require a specific value when using [Icon](/web/icon) with [Button](/web/button). Gestalt is more permissive than we recommend internally for adding icons to Buttons, so Buttons using \`iconEnd\` must use the &quot;arrow-down&quot; icon and Buttons with link role using \`iconEnd\` must use the &quot;visit&quot; icon.
```
Comment (dangerismycat) [inline docs/pages/get_started/developers/eslint_plugin.js:91]: nit: the user isn't actually using our Icon button component here, so no need to add the link

File: docs/pages/web/button.js [hunk 2]
```diff
@@ -523,10 +525,19 @@ function SemiTransparentWhiteButtonExample() {
...
           description={`
-Icon end
-Adds an icon after the Button text. Icons should only be used to visually reinforce a specific function or interaction of the Button. Menus and external links are a common use case. Use \`visit\` when linking to an external URL or \`arrow-down\` when displaying a Popover on click. Note that iconEnd on Button is not accessible to screen readers.
+\`iconEnd\` adds an icon after the Button text. Icons should only be used to visually reinforce a specific function or interaction of the Button. Menus and external links are a common use case. Use \`visit\` when linking to an external URL or \`arrow-down\` when displaying a Popover on click. Note that iconEnd on Button is not accessible to screen readers.
 `}
         >
+          <SlimBanner
+            type="recommendationBare"
+            iconAccessibilityLabel="Recommendation"
+            message="Use Gestalt's Eslint rule to enforce the correct icons usage in Button."
```
Comment (dangerismycat) [inline docs/pages/web/button.js:534]: nit: according to [their docs](https://eslint.org/), it's usually styled as "ESLint"

File: packages/eslint-plugin-gestalt/src/button-icon-restrictions.test.js [hunk 3]
```diff
@@ -20,7 +20,17 @@ const invalidWrongIcon = readFileSync(
...
-const errorMessage = 'Buttons using iconEnd must use "arrow-down"';
+const invalidWrongIconLinkRole = readFileSync(
+  path.resolve(
+    __dirname,
+    './__fixtures__/button-icon-restrictions/invalid/invalid-wrong-link-role-icon.js',
+  ),
+  'utf-8',
+);
+
+const errorMessage = 'Buttons using iconEnd must use "arrow-down" icon';
```
Comment (dangerismycat) [inline packages/eslint-plugin-gestalt/src/button-icon-restrictions.test.js:31]: nit: you can save yourself some typing by exporting the error messages from the rule file

**Review by dangerismycat — APPROVED** [review]

File: docs/pages/get_started/developers/eslint_plugin.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/get_started/developers/eslint_plugin.js:91]: DONE

File: packages/eslint-plugin-gestalt/src/button-icon-restrictions.test.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/eslint-plugin-gestalt/src/button-icon-restrictions.test.js]: DONE

**Pull request merged**